### ISSUE TITLE
Revisit reccomended license (and reccomending practices)

### ISSUE THREAD

**Issue by dcwalk** [edgi-govdata-archiving/overview#76]
We've had some questions around our approach to licensing. The decision coming out of Toronto's event was to GPLv3 everything and recommend that going forward. 

Given the repo shake up some of those recommendations are buried/lost (were associated with event planning).

How do we want to address this as a community?
My plan was to have a discussion on `#dev` and reference at our Standup

**Comment by Mr0grog** [comment]
I’m generally biased towards the BSD 3-clause or MIT licenses, mainly because they are:

- Open to any sort of use and participation (in particular, I think it is generally good to invite industry participation by making it non-worrisome for private entities; for most code, I don’t find myself concerned with whether some people might use it to make money)
- Straightforward and clear

I’m not usually a huge fan of the GPL, mainly because it creates barriers for participation, but also because of the continued confusion it has offered in terms of what counts as “linking” and what uses, exactly, are covered. In general, it’s anathema to use by private companies that develop and sell software, which means you lose a lot of potential contributors (and/or time) from those companies.

I *do* think the GPL is great for code that is explicitly educational, where the source itself (as opposed the the software it forms) is intended as a public resource. Projects like EDGI software, which aim to act as a public resource (but are not explicitly educational), kind of fit here, too, but I’m not sure there’s huge value in preventing someone from extending that public resource just because they might sell the result.

In short, I’d lean more towards BSD 3-clause over GPL, but that leaning is not especially strong.

**Comment by dcwalk** [comment]
Thanks for the comments @Mr0grog...

In the chat we had a few people chime in for GPLv3, which is what we had previous agreed upon, a good reminder to copy out those thoughts here:
> I have a strong personal preference for GPL3, especially for work of this nature. I think at the very least you’d want a license that grants contributors patent rights to the parent organization to avoid any mischief on that front (e.g Apache v2). 

> +1 GPL generally and AGPL for code that is only distributed over a network

**Comment by Mr0grog** [comment]
Heh, yeah, I didn’t see the Slack discussion until just after I made the comment, at which point my notes felt kinda redundant 😅

**Comment by dcwalk** [comment]
Not redundant! Thanks for the input, it is good to get a sense from a range of people

**Comment by ebarry** [comment]
To clarify, there's no barrier to companies extending and also selling
their extension to a public (GPL) resource, it's that they can't enclose /
remove their work from the public.

--

+1 336-269-1539
@lizbarry <http://twitter.com/lizbarry>

**Comment by Mr0grog** [comment]
> To clarify, there's no barrier to companies extending and also selling their extension to a public (GPL) resource, it's that they can't enclose/remove their work from the public.

They *do* have to additionally make the source code of that work (which often comprises the majority of the work) available for free, though. It may not be a technical barrier, but most companies certainly view as one—it has the same affect on corporate behavior in most cases. (For example, as a former employee of a large computer company, I’ve had my share of unpleasant interactions with management and legal teams about this—even with off-hours, own-equipment contribution.)

But anyway! I have no problem with EDGI projects I’m working on here using the GPL. I think the arguments for doing so on these projects is more than reasonable.

**Comment by dcwalk** [comment]
This was addressed in #92, closing for now!
